app/pages/multisig/updates/AddIdentityProvider/CreateAddIdentityProvider.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useCallback } from 'react';
+import { Validate, ValidationRule } from 'react-hook-form';
+
+import { EqualRecord, AddIdentityProvider, Description } from '~/utils/types';
+import { isHex, onlyDigitsNoLeadingZeroes } from '~/utils/basicHelpers';
+import Form from '~/components/Form';
+import { UpdateProps } from '~/utils/transactionTypes';
+import { useIdentityProviders } from '~/utils/dataHooks';
+
+export type AddIdentityProviderFields = Omit<
+    AddIdentityProvider,
+    'ipDescription'
+> &
+    Description;
+
+const cdiKeyLength = 64;
+
+const fieldNames: EqualRecord<AddIdentityProviderFields> = {
+    name: 'name',
+    url: 'url',
+    description: 'description',
+    ipIdentity: 'ipIdentity',
+    ipVerifyKey: 'ipVerifyKey',
+    ipCdiVerifyKey: 'ipCdiVerifyKey',
+};
+
+export const fieldDisplays = {
+    name: 'Name',
+    url: 'URL',
+    description: 'Description',
+    ipIdentity: 'Identity Provider',
```

Comment:
Isn't this more like the 'Identity Provider index' or something along those lines? It looks a bit weird when it asks you to:
```
Enter Identity Provider here
```

Reply:
Hmm, I choose this to be consistent with the ledger, which also displays "Identity Provider".